doc/src/manual/performance-tips.md

```diff
@@ -485,6 +485,26 @@ annotation in this context in order to achieve type stability. This is because t
 cannot deduce the type of the return value of a function, even `convert`, unless the types of
 all the function's arguments are known.
 
+Type annotation will not enhance (and can actually hinder) performance if the type is constructed
+at run-time.  This is because the compiler cannot use the annotation to specialize the subsequent
+code, and the type-check itself takes time.  For example, in the code:
+
+```julia
+function nr(a, prec)
+    ctype = prec == 32? Float32 : Float64
```

Comment:
note that ternaries without spaces on both sides of `?` are deprecated on master, so we should avoid the deprecated syntax in doc examples